client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -415,9 +450,15 @@ impl GenericProto {
 		protocol_name: Cow<'static, [u8]>,
 		message: impl Into<Vec<u8>>,
 	) {
-		if !self.is_open(target) {
-			return;
-		}
+		let conn = match self.peers.get(target).and_then(|p| p.get_open()) {
+			None => {
+				debug!(target: "sub-libp2p",
+					"Tried to sent notification to {:?} without an open channel.",
```

Comment:
What differentiates a channel from a substream? If they are interchangeable, could we settle on one term?

Reply:
"channel" is the more abstract term, i.e. could be a connection or a substream. I don't really mind about the terminology in debug logs, I just thought that the behaviour shouldn't really know / care about the distinction, i.e. if a connection handler reports "open", then the behaviour knows there is an "open (communication) channel" to the peer via that connection, whatever that means in terms of substreams.